### ISSUE TITLE
Allow class to be used for function

### ISSUE THREAD

**Issue by Daniel15** [Daniel15/fdk-dotnet#7]
Currently we use:
```csharp
FdkHandler.Handle(async (ctx, input) => 
{
    // Do stuff
});
```

ASP.NET Core allows middleware to be defined either as a lambda function (similar to what we do here), or as a class. When using a class, it handles injecting dependencies both through the constructor (for singletons) and through the method call itself (for transient and scoped dependencies)

This would potentially allow for a cleaner API, where we can just determine whether the input is a string, a stream, or JSON just based on the function's argument type, rather than having to call `AsString` etc. explicitly.

Potential API:
```csharp
public class MyFunction
{
    // Example of dependency injected via constructor
    private readonly ILogger<MyFunction> _logger;
    public MyFunction(ILogger<MyFunction> logger)
    {
        _logger = logger;
    }

    // Input is automatically passed in as a string
    // `IFoo` could be some scoped dependency
    public Task<string> InvokeAsync(string input, IFoo foo)
    {
        return "Hello " + input;
    }
}
```

This looks similar to what the Java FDK does.

In `Program.cs`, we could have an init API like this:
```csharp
FdkHandler.Handle<MyFunction>();
```

---

Code reference for how middleware handles it: https://github.com/aspnet/AspNetCore/blob/master/src/Http/Http.Abstractions/src/Extensions/UseMiddlewareExtensions.cs#L145

**Comment by Daniel15** [comment]
Started working on this, basic pieces are there in master, but it's not complete yet.

**Comment by Daniel15** [comment]
TODO: Inject `CancellationToken`

**Comment by Daniel15** [comment]
Implemented in 1.0.0 release.
